src/python/utils/diff_statistics.py

```diff
@@ -0,0 +1,167 @@
+from pathlib import Path
+from typing import Union, Set
+
+from loguru import logger
+
+from src.python.repo import Repository
+
+
+def get_diff(
+    repo: Repository, left_sha: str, right_sha: str, diff_log_file: Union[None, Path]
+) -> str:
+    """
+    Computes the diff between two branches using git diff.
+    Args:
+        repo (Repository): The repository object.
+        left_sha (str): The left sha.
+        right_sha (str): The right sha.
```

Comment:
There is still a consistency between "branches" on line 13 and "SHA" on lines 16 and 17.  The same discrepancy appears elsewhere too.